csharp/private/macros/setup_basic_nuget_package.bzl

```diff
@@ -0,0 +1,94 @@
+load("//csharp/private:providers.bzl", "CSharpAssembly")
+load("//csharp/private:rules/imports.bzl", "import_library", "import_multiframework_library")
+
+def _import_dll(dll, has_pdb, imports):
+    path = dll.split("/")
+
+    tfm = path[1]
+
+    # Ignore frameworks we don't support (like net35)
+    if tfm not in CSharpAssembly:
+        return
+
+    lib_name = path[-1].rsplit(".", 1)[0]
+
+    target_name = "%s-%s" % (lib_name, tfm)
+
+    if lib_name not in imports:
+        imports[lib_name] = {tfm: target_name}
+    else:
+        imports[lib_name][tfm] = target_name
+
+    if dll in has_pdb:
+        import_library(
+            name = target_name,
+            target_framework = tfm,
+            dll = dll,
+            pdb = dll[:-3] + "pdb",
+        )
+    else:
+        import_library(
+            name = "%s-%s" % (lib_name, tfm),
+            target_framework = tfm,
+            dll = dll,
+        )
+
+def setup_basic_nuget_package():
+    """This macro gets used to implement the default NuGet BUILD file.
+
+       We are limited by the fact that Bazel does not allow the analysis phase to
+       read the contents of source files, e.g. to correctly configure deps. For
+       more advanced usages a BUILD file will need to be generated outside of
+       Bazel.
+
+       This has to be public so that packages can call it but you probably
+       shouldn't use it directly.
+    """
+    dlls = native.glob(["**/*.dll"])
```

Comment:
Globs are relative to the caller?

Reply:
Probably don't need to care right away, but what does a failure look like if one of these is say, a directory?

Reply:
Will this pick up refdlls? Should we only be looking in `lib`?

Reply:
Oh shoot I had this as `lib/*/*.dll` at one point but was messing around. Will fix.